### PR TITLE
ENH: pass legend_kwds to colorbar when relevant

### PR DESCRIPTION
**Pull request by rbuffat**
This PR exposes the kwargs of matplotlib.pyplot.colorbar to plot().  The following example demonstrates its use for horizontal colorbars located at the bottom.

```
from mpl_toolkits.axes_grid1.axes_divider import make_axes_locatable
import geopandas as gp
import matplotlib.pyplot as plt

df = gp.read_file("g2g14vz.shp")


cmap = "inferno"

fig, (ax1, ax2) = plt.subplots(1, 2)

# With horizontal colorbar
divider = make_axes_locatable(ax1)
cax1 = divider.append_axes("bottom", size="5%", pad=0)

df.plot(column="AREA_HA", cmap=cmap, ax=ax1, cax=cax1, legend=True, cbar_kwds={"orientation": "horizontal"}, vmin=0, vmax=2000)

# With vertical colorbar
divider2 = make_axes_locatable(ax2)
cax2 = divider2.append_axes("bottom", size="5%", pad=0)

df.plot(column="AREA_HA", cmap=cmap, ax=ax2, cax=cax2, legend=True, vmin=0, vmax=2000)

plt.savefig("colorbar_test.png")
```
colorbar_test

### CODE REVIEW

File: geopandas/plotting.py [hunk 1]
```diff
@@ -484,10 +487,13 @@ def plot_dataframe(df, column=None, cmap=None, color=None, ax=None, cax=None,
         plot_linestring_collection(ax, lines, values[line_idx],
                                    vmin=mn, vmax=mx, cmap=cmap, **style_kwds)
 
+    if cbar_kwds is None:
+        cbar_kwds = {}
+
     if cax is not None:
-        cbar_kwargs = {"cax": cax}
+        cbar_kwds.update({"cax": cax})
```
Comment (rbuffat) [inline geopandas/plotting.py]: alternatively setdefault() could be used instead of update()

**Comment by martinfleis** [comment]
Hi @rbuffat,

thanks for this work! I think I would prefer passing colorbar keywords using existing `legend_kwds`. Because a plot will never have both categorical legend and colorbar. And to me, colorbar is a legend. There is already a lot of arguments available for plot, so I would prefer not adding more if not necessary. 

Also, can you try to make some tests? And ideally example to docs?

It is much welcome PR though. (#697, #961). Thanks!

File: geopandas/plotting.py [hunk 2]
```diff
@@ -376,7 +376,9 @@ def plot_dataframe(df, column=None, cmap=None, color=None, ax=None, cax=None,
         Size of the resulting matplotlib.figure.Figure. If the argument
         axes is given explicitly, figsize is ignored.
     legend_kwds : dict (default None)
-        Keyword arguments to pass to ax.legend()
+        Keyword arguments to pass to matplotlib.pyplot.legend() in case of legend=True and
```
Comment (martinfleis) [inline geopandas/plotting.py]: This is true from our point of view, but you can also do `gdf.plot('data', scheme='quantiles', legend=True)`, which sets categorical to True, but user did not touch it so they might expect it defaulting to False. Can cause confusion. I would use simpler language here and say something along "Keyword arguments to pass to matplotlib.pyplot.legend() or matplotlib.pyplot.colorbar()".

**Comment by rbuffat** [comment]
@martinfleis Creating a test is more difficult than anticipated. I did not found a way to access the colorbar object from the axes or the figure using matplotlib's public interface. My knowledge of matplotlib's internal behavior is limited,  maybe you have an idea how to test this?

One solution would be that plot returns the legend/colorbar object in case a legend is created.  This has the benefit of having the possibility to set a label to the colorbar. However, I think this could potentially break existing code.

**Comment by martinfleis** [comment]
@rbuffat some of the tests access colorbar using `ax.get_figure().axes[1].collections[0]` in https://github.com/geopandas/geopandas/blob/a441237ae1d73edf540db2f761dda94be593fc81/geopandas/tests/test_plotting.py#L138
That might help?

**Comment by rbuffat** [comment]
I saw this test, but like I said I'm not familiar enough with matplotlib's internals to see at the moment a good way to test this without accessing the result of colorbar().

Matplotlib uses tests that compare images. E.g. https://github.com/matplotlib/matplotlib/blob/33376e942a2f61bcfc711720eab34dcb1ef6cf18/lib/matplotlib/tests/test_colorbar.py#L229

**Comment by rbuffat** [comment]
ok, maybe I found a way of testing it. I was wrong, it is not only possible to add a label to a colorbar with set_label on the colorbar object, but also with the kwargs (this is not documented in the matplotlib docs). Thus we can test for the labels of the axes that is created through colorbar().

**Comment by martinfleis** [comment]
Nice! Good job. One final request - can you add an example to User guide? Making maps/Creating a legend so we have it properly documented? Just this + explanation:

`world.plot(column='pop_est', legend=True, legend_kwds={'label': 'Population', "orientation": "horizontal"})`

Thank you.

**Comment by martinfleis** [comment]
Once merged it will also close #741 as it is a duplicated (unfinished) PR.

**Comment by rbuffat** [comment]
@martinfleis  Is this something you had in mind? I do unfortunately not yet know how to render the rst files locally.

**Comment by martinfleis** [comment]
@rbuffat Yes, thank you! I just fixed missing quotation marks. Waiting for tests to complete and merging. Good job.

**Comment by rbuffat** [comment]
@martinfleis @jorisvandenbossche Thanks for merging!

**Comment by njguerram** [comment]
> Nice! Good job. One final request - can you add an example to User guide? Making maps/Creating a legend so we have it properly documented? Just this + explanation:
> 
> `world.plot(column='pop_est', legend=True, legend_kwds={'label': 'Population', "orientation": "horizontal"})`
> 
> Thank you.

It worked, but how could I to set the label size? 'labelsize' does not work.

**Comment by jcarvoli** [comment]
Hello, thanks for your work! Same question as njguerram, how could we additionally change the label fontsize?

**Pull request merged**